extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/impl/sink/ShardCountMonitorImpl.java

```diff
@@ -42,35 +47,26 @@
      */
     private static final double RATIO_OF_DESCRIBE_STREAM_RATE_UTILIZED = 0.1;
 
-    private final AtomicInteger shardCount;
```

Comment:
the reason this was an AtomicInteger was to share the shard count with other processors I guess. So only one of the processors (first one) runs periodically and sets the shard count (to this AtomicInteger) and other processors (those having a noop monitor) just gets the shard count from that AtomicInteger.

With this change other processors will see shard count as 0 all the time

Reply:
Ah, true. So I broke it... I couldn't make sense of the code.

Reply:
Fixed